[0031] Org-Wide Rust Style Guide

nrdxp opened the pull request:
[rendered](https://github.com/ekala-project/eeps/blob/rust-style/eeps/0031-org-wide-rust-style-guide.md)

nrdxp committed:
init(31): Org-Wide Rust Style Guide

nrdxp committed:
refine: explicate more details from the style guide

Include much of the details from eka's style guide directly in the EEP
to be a bit more self-contained.

jonringer commented on eeps/0031-org-wide-rust-style-guide.md:
```diff
@@ -0,0 +1,218 @@
+
+# Acknowledgements
+
+Thanks to the original authors of the `STYLE_GUIDE.md` for establishing a strong foundation for this proposal. Special thanks to @jonringer for his ongoing efforts in eka-ci and in driving ekapkgs forward.
```
<3

jonringer reviewed (approved):
I'm not really opinionated, and this looks reasonable.

nrdxp committed:
[0031]: add justification for rustfmt settings

nrdxp commented:
one setting of the rustfmt config in particular that might warrant some further scrutiny is: `imports_granularity = "Module"` ([doc](https://rust-lang.github.io/rustfmt/?version=v1.8.0&search=#imports_granularity))

Tried to strike a balance from the default `Crate` (which is honestly very cluttered and ugly), and the `Item` which might be beneficial for minimal diffs, but is also quite verbose. However, maybe minimal diffs should be the priority? Just something to consider.

nrdxp committed:
[0031]: minor clarification to sort order

The easiest way to enforce sorting is with LLMs, but without very
explicit instructions, they get confused and produced mixed results on
the order of various types. For that reason, we shall be very explicit
about how to sort types to alleviate this ambiguity and hopefully ease
automation.

jonringer commented on eeps/0031-org-wide-rust-style-guide.md:
```diff
@@ -0,0 +1,292 @@
+
+All other top-level items—including modules, constants, traits, and functions—must be sorted alphabetically by their identifier.
+
+Types (`struct`, `enum`, and `type` aliases) must be sorted as a single, interlaced group, alphabetically by their identifier.
+
```
what's next? fascism?

lol

nrdxp replied:
I mean in practice it's never going to be perfect, but it's just a helpful guideline to try to maintain some semblance of organization in large modules.
